src/gt4py/next/program_processors/runners/dace_fieldview/transformations/loop_blocking.py

```diff
@@ -0,0 +1,607 @@
+                    blocking_parameter=blocking_parameter,
+                    independent_nodes=independent_nodes,
+                    state=state,
+                    sdfg=sdfg,
+                )
+
+                # Check if the partition exists.
+                if class_res is None:
+                    return None
+                if class_res is True:
+                    found_new_independent_node = True
+
+            # If we found a new independent node then we have to continue.
+            if not found_new_independent_node:
+                break
+
+        # After the independent set is computed compute the set of dependent nodes
+        #  as the set of all nodes adjacent to `outer_entry` that are not dependent.
+        dependent_nodes: set[dace_nodes.Node] = {
+            edge.dst for edge in state.out_edges(outer_entry) if edge.dst not in independent_nodes
+        }
+
+        return (independent_nodes, dependent_nodes)
+
+    def classify_node(
+        self,
+        node_to_classify: dace_nodes.Node,
+        outer_entry: dace_nodes.MapEntry,
+        blocking_parameter: str,
+        independent_nodes: set[dace_nodes.Node],
+        state: dace.SDFGState,
+        sdfg: dace.SDFG,
+    ) -> bool | None:
+        """Internal function for classifying a single node.
+
+        The general rule to classify if a node is independent are:
+        - The node must be a Tasklet or an AccessNode, in all other cases the
+            partition does not exist.
+        - `free_symbols` of the nodes shall not contain the `blocking_parameter`.
+        - All incoming _empty_ edges must be connected to the map entry.
+        - A node either has only empty Memlets or none of them.
+        - Incoming Memlets does not depend on the `blocking_parameter`.
+        - All incoming edges must start either at `outer_entry` or at dependent nodes.
+        - All output Memlets are non empty.
+
+        Returns:
+            The function returns `True` if `node_to_classify` is considered independent.
+            In this case the function will add the node to `independent_nodes`.
+            If the function returns `False` the node was classified as a dependent node.
+            The function will return `None` if the node can not be classified, in this
+            case the partition does not exist.
+
+        Args:
+            node_to_classify: The node that should be classified.
+            outer_entry: The entry of the map that should be partitioned.
+            blocking_parameter: The iteration parameter that should be blocked.
+            independent_nodes: The set of nodes that was already classified as
+                independent, in which case it is added to `independent_nodes`.
+            state: The state containing the map.
+            sdfg: The SDFG that is processed.
+        """
+
+        # We are only able to handle certain kind of nodes, so screening them.
+        if isinstance(node_to_classify, dace_nodes.Tasklet):
```

Comment:
Don't need to change anything here, just a comment for the future: I recommend to use structural pattern matching for this kind of situations where several conditional branches depend on the type of some value (e.g.: 
https://stackoverflow.com/questions/67524641/convert-multiple-isinstance-checks-to-structural-pattern-matching )